src/shared/components/image-button/ImageButton.tsx

```diff
@@ -24,52 +28,57 @@ import Tooltip from 'src/shared/components/tooltip/Tooltip';
 
 import { Status } from 'src/shared/types/status';
 
-import imageButtonStyles from './ImageButton.scss';
+import styles from './ImageButton.scss';
 
 export type ImageButtonStatus =
   | Status.DEFAULT
   | Status.SELECTED
   | Status.UNSELECTED
   | Status.DISABLED;
 
-export type Props = {
-  status: ImageButtonStatus;
-  description: string;
+type ChildlessButtonProps = Omit<
+  ButtonHTMLAttributes<HTMLButtonElement>,
+  'children'
+>;
+
+export type Props = ChildlessButtonProps & {
+  status?: ImageButtonStatus;
+  description?: string;
   image: FunctionComponent | string;
-  className?: string;
-  statusClasses?: { [key in ImageButtonStatus]?: string };
-  onClick?: (event?: React.MouseEvent<HTMLDivElement>) => void;
 };
 
 export const ImageButton = (props: Props) => {
+  const {
+    status = Status.DEFAULT,
+    description,
+    image: Image,
+    className: classNameFromProps,
+    ...otherProps
+  } = props;
   const [hoverRef, isHovered] = useHover<HTMLDivElement>();
 
-  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
-    props.onClick && props.onClick(event);
-  };
-
-  const buttonProps =
-    props.status === Status.DISABLED ? {} : { onClick: handleClick };
-
-  const styles = props.statusClasses
-    ? { ...imageButtonStyles, ...props.statusClasses }
-    : imageButtonStyles;
-
   const imageButtonClasses = classNames(
-    imageButtonStyles.imageButton,
-    props.className,
-    styles[props.status]
+    styles.imageButton,
+    styles[status],
+    classNameFromProps
   );
 
-  const shouldShowTooltip = Boolean(props.description) && isHovered;
+  const shouldShowTooltip = Boolean(description) && isHovered;
 
+  // NOTE: the only reason to wrap the button in a div element is so that we could react in a custon manner
+  // to a hover event over a disabled button (i.e. show the tooltip).
+  // If we were satisfied with just the `title` attribute, we wouldn't need this wrapper
   return (
-    <div ref={hoverRef} className={imageButtonClasses} {...buttonProps}>
-      <button>
+    <div ref={hoverRef} className={styles.wrapper}>
+      <button
+        className={imageButtonClasses}
+        {...otherProps}
+        disabled={status === Status.DISABLED}
+      >
         {typeof props.image === 'string' ? (
           <img src={props.image} alt={props.description} />
```

Comment:
should this be:
`<img src={Image} alt={description} />`

Reply:
`src={Image}`, with capital-case `Image` feels wrong. Let it be as it is.